dvc/repo/plots/__init__.py

```diff
@@ -260,6 +302,168 @@ def _collect_plots(
     return result
 
 
+def _get_data_targets(pathlib, definitions: Dict):
+    result: Set = set()
+    if "data" in definitions:
+        for file_relpath, content in definitions["data"].items():
+            if "data" in content:
+                for plot_id, config in content["data"].items():
+                    result = result.union(
+                        infer_data_sources(
+                            pathlib, file_relpath, plot_id, config
+                        )
+                    )
+    return result
+
+
+def infer_data_sources(pathlib, config_path, plot_id, config):
+    import posixpath
+
+    def _deduplicate(lst: List):
+        return list({elem: None for elem in lst}.keys())
+
+    y = config.get("y", None)
+    if isinstance(y, dict):
+        sources = list(y.keys())
+    else:
+        sources = [plot_id]
+
+    # TODO bug in dvc_objects? normpath normalizes to windows path on windows
+    return _deduplicate(
+        posixpath.normpath(pathlib.join(pathlib.dirname(config_path), source))
```

Comment:
Why is this needed?

Reply:
DvcFileSystem uses `posixpath` style paths, regardless the fs its on. For some reason, `join` and `dirname` works on Windows with the posixpaths, but the `normpath` normalizes into Windows style path, which results in problems of evaluating data sources and matching the renderers down the road. It seems like inconsistent behavior, hence TODO to investigate.